homeassistant/components/mqtt/device_trigger.py

```diff
@@ -267,12 +301,44 @@ async def async_setup_trigger(
 ) -> None:
     """Set up the MQTT device trigger."""
     config = TRIGGER_DISCOVERY_SCHEMA(config)
-    device_id = update_device(hass, config_entry, config)
+
+    # Check if there is an existing conflicting trigger first
```

Comment:
I don't understand, why do we need to iterate over the device registry? All triggers are in mqtt_data.device_triggers.

Reply:
Added a comment and a test